server/server.go

```diff
@@ -187,11 +195,81 @@ func (s *Server) Serve() error {
 	if err := s.exportServices(); err != nil {
 		return err
 	}
+	if err := s.exportInternalServices(); err != nil {
+		return err
+	}
 	metadata.ExportMetadataService()
 	registry_exposed.RegisterServiceInstance(s.cfg.Application.Name, s.cfg.Application.Tag, s.cfg.Application.MetadataType)
 	select {}
 }
 
+// In order to expose internal services
+func (s *Server) exportInternalServices() error {
+	cfg := &ServiceOptions{}
+	cfg.Application = s.cfg.Application
+	cfg.Provider = s.cfg.Provider
+	cfg.Protocols = s.cfg.Protocols
+	cfg.Registries = s.cfg.Registries
+
+	services := make([]*InternalService, 0, len(proServices))
+
+	proLock.Lock()
+	defer proLock.Unlock()
+	for _, service := range proServices {
+		if service.Init == nil {
+			return errors.New("internal service init func is empty, please set the init func correctly ")
+		}
+		sd, ok := service.Init(cfg)
+		if !ok {
+			continue
+		}
+		newSvcOpts, err := s.genSvcOpts(sd.Handler, sd.Opts...)
+		if err != nil {
+			return err
+		}
+		service.svcOpts = newSvcOpts
+		service.info = sd.Info
+		services = append(services, service)
+	}
+
+	sort.Slice(services, func(i, j int) bool {
+		return services[i].Priority < services[j].Priority
+	})
+
+	for _, service := range services {
+		if service.BeforeExport != nil {
+			service.BeforeExport(service.svcOpts)
+		}
+		err := service.svcOpts.ExportWithInfo(service.info)
+		if service.AfterExport != nil {
+			service.AfterExport(service.svcOpts, err)
+		}
+		if err != nil {
+			logger.Errorf("export %s internal service failed, err: %s", service.svcOpts.Service.Interface, err)
+			return err
+		}
+	}
+
+	return nil
+}
+
+// InternalService for dubbo internal services
+type InternalService struct {
+	svcOpts *ServiceOptions
+	info    *ServiceInfo
+	// This is required
+	// Return serviceDefinition and bool, where bool indicates whether it is exported
+	Init         func(options *ServiceOptions) (*ServiceDefinition, bool)
+	BeforeExport func(options *ServiceOptions)
```

Comment:
This options is InternalService.svcOpts itself or other ServiceOptions would be passed in?